### DISCUSSION TITLE
Windows version

### DISCUSSION THREAD

**Discussion by burdiyan** [pkgxdev/pkgx#257] [Ideas]
I'm super excited about tea. I've been dreaming about a universal package manager for years. It seems ridiculous that the world is so fragmented, but most package managers and build systems end up doing so similar things, but vastly incompatible.

For the past 3 years I've been working on a project that targets Linux, macOS, and Windows. And we also want to support all those platforms as a build environment (meaning that you could work on those platforms to work on the project, and build it for that particular problem).

We've improved the developer experience quite a bit on Linux and macOS by using [Nix package manager](https://nixos.org), but it's not available for Windows. As far as I understand (by looking into the installer script) tea doesn't support Windows either.

P.S. +1 for your "Don't use Docker" statement in the readme :) I built a decent development environment years ago with Docker for a company I've been working at, but it still sucked :) Since I discovered Nix, I'm silently laughing when people are trying to do the same, or claiming that using Docker for development is optimal, or even good-enough.

So, +100 for a Windows version of tea :) And I don't mean WSL, BTW :) You can't use the Windows compiler from WSL, so you can't really compile a Windows GUI app from inside WSL (at least without losing your mind with hackery).

You might require [MSYS2](https://www.msys2.org) or something like that for tea to work on Windows, at least to be able to run shell scripts without rewriting them to PowerShell.

**Comment by mxcl** [comment]
Hi! Great to hear all these things, couldn't agree more that it's silly how duplicated everything is in this space! We hope we can help fix that.

We’re actively working on the Windows native version as we speak. We hoped to launch with it but we couldn't get there.

We'll check out MSYS2 since maybe that's a missing piece in our efforts.

**Reply by ykp01** [reply]
sadece çay istiyorum 😌

**Reply by brad-jones** [reply]
It's halfway through 2023, is there an update to the Windows story that you can share with us? Even if it didn't support the magic it would be game changing.

**Reply by mxcl** [reply]
yeah sorry. This is something I really want but I haven't time to work on it. It will be easier now that libtea and tea/cli are split out as we can start by getting libtea tests to pass on windows.

**Reply by brad-jones** [reply]
Oh my libtea how awesome. This thing just keeps getting better & better.
